package/src/components/Poll/CreatePollContent.tsx

```diff
@@ -1,214 +1,126 @@
+import { CreatePollOptions, CurrentOptionPositionsCache } from './components';
+
+import { CreatePollHeader } from './components/CreatePollHeader';
+import { MultipleAnswersField } from './components/MultipleAnswersField';
+import { NameField } from './components/NameField';
 
 import {
   CreatePollContentContextValue,
   CreatePollContentProvider,
   InputMessageInputContextValue,
-  useChatContext,
   useCreatePollContentContext,
   useTheme,
   useTranslationContext,
 } from '../../contexts';
-import { SendPoll } from '../../icons';
-
-export const isMaxNumberOfVotesValid = (maxNumberOfVotes: string) => {
-  const parsedMaxNumberOfVotes = Number(maxNumberOfVotes);
+import { useMessageComposer } from '../../contexts/messageInputContext/hooks/useMessageComposer';
+import { useStateStore } from '../../hooks/useStateStore';
+
+const pollComposerStateSelector = (state: PollComposerState) => ({
+  allow_answers: state.data.allow_answers,
+  allow_user_suggested_options: state.data.allow_user_suggested_options,
+  enforce_unique_vote: state.data.enforce_unique_vote,
+  max_votes_allowed: state.data.max_votes_allowed,
+  name: state.data.name,
+  options: state.data.options,
+  voting_visibility: state.data.voting_visibility,
+});
 
-  return (
-    !isNaN(parsedMaxNumberOfVotes) && parsedMaxNumberOfVotes > 1 && parsedMaxNumberOfVotes <= 10
-  );
-};
+export const POLL_OPTION_HEIGHT = 71;
 
 export const CreatePollContent = () => {
   const { t } = useTranslationContext();
-  const [pollTitle, setPollTitle] = useState('');
-  const [pollOptions, setPollOptions] = useState<PollOptionData[]>([{ text: '' }]);
-  const [multipleAnswersAllowed, setMultipleAnswersAllowed] = useState(false);
-  const [maxVotesPerPerson, setMaxVotesPerPerson] = useState('');
-  const [maxVotesPerPersonEnabled, setMaxVotesPerPersonEnabled] = useState(false);
-  const [isAnonymous, setIsAnonymous] = useState(false);
-  const [optionSuggestionsAllowed, setOptionSuggestionsAllowed] = useState(false);
-  const [commentsAllowed, setCommentsAllowed] = useState(false);
-  const [duplicates, setDuplicates] = useState<string[]>([]);
 
-  const { closePollCreationDialog, createAndSendPoll } = useCreatePollContentContext();
+  const messageComposer = useMessageComposer();
+  const { pollComposer } = messageComposer;
+  const { allow_answers, allow_user_suggested_options, options, voting_visibility } = useStateStore(
+    pollComposer.state,
+    pollComposerStateSelector,
+  );
+
+  const { createPollOptionHeight, closePollCreationDialog, createAndSendPoll } =
+    useCreatePollContentContext();
 
   // positions and index lookup map
   // TODO: Please rethink the structure of this, bidirectional data flow is not great
   const currentOptionPositions = useSharedValue<CurrentOptionPositionsCache>({
-    inverseIndexCache: { 0: 0 },
-    positionCache: { 0: { updatedIndex: 0, updatedTop: 0 } },
+    inverseIndexCache: {},
```

Comment:
Did we decide that we don't have any options initially ? I know we implemented these changes on a call, but I forgot what the reasoning was

Reply:
We will, anyway, initialise the value to a valid option in the useEffect next to it, so I decided to remove it.